src/devices/MotorHat/MotorPinProvider.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+
+namespace Iot.Device.MotorHat
+{
+    /// <summary>
+    /// Represents a provider for <see cref="MotorPins"/>
+    /// </summary>
+    public interface IMotorPinProvider
+    {
+        /// <summary>
+        /// Gets the <see cref="MotorPins"/> for motor at the specified <paramref name="index"/>
+        /// </summary>
+        /// <param name="index">The index of the motor to get pins for</param>
+        /// <returns></returns>
+        public abstract MotorPins GetPinsForMotor(int index);
+    }
+
+    /// <summary>
+    /// Abstract base class for creating DCMotor instances
+    /// </summary>
+    public static class MotorPinProvider
+    {
+        /// <summary>
+        /// <see cref="IMotorPinProvider"/> for AdaFruit motor hat
+        /// </summary>
+        public static readonly IMotorPinProvider AdaFruit = new AdafruitMotorPinProvider();
+
+        /// <summary>
+        /// <see cref="IMotorPinProvider"/> for Aliexpress motor hat
+        /// </summary>
+        public static readonly IMotorPinProvider Aliexpress = AdaFruit;
+
+        /// <summary>
+        /// <see cref="IMotorPinProvider"/> for Waveshare motor hat
+        /// </summary>
+        public static readonly IMotorPinProvider Waveshare = new WaveshareMotorPinProvider();
+
+        /// <summary>
+        /// Default <see cref="IMotorPinProvider"/>
+        /// </summary>
+        public static readonly IMotorPinProvider Default = AdaFruit;
+    }
+
+    /// <summary>
+    /// <see cref="MotorPinProvider"/> implementation for AdaFruit/Aliexpress
+    /// </summary>
+    /// <remarks>
+    /// These correspond to motor hat screw terminals M1, M2, M3 and M4.
+    /// </remarks>
+    public class AdafruitMotorPinProvider : IMotorPinProvider
+    {
+        /// <summary>
+        /// Gets the <see cref="MotorPins"/> for motor at the specified <paramref name="index"/>
+        /// </summary>
+        /// <param name="index">The index of the motor to get pins for</param>
+        /// <returns></returns>
+        public MotorPins GetPinsForMotor(int index)
```

Comment:
you most likely can use /// <inheritsdoc /> instead. Same for the other class. As the comment is the same.

Reply:
I assumed you meant `/// <inheritdoc/>` which is what I've done.